WordPress/Classes/ViewRelated/Post/PostListViewController.swift

```diff
@@ -494,7 +494,23 @@ class PostListViewController: AbstractPostListViewController, UIViewControllerRe
     }
 
     func cell(_ cell: UITableViewCell, handleTrashPost post: AbstractPost) {
-        deletePost(post)
+        if (post.status == "trash") {
+            weak var weakSelf = self
```

Comment:
Instead of this, you can declare `weakSelf` as part of your delete action later on, like so:

`{ [weak self] action in`

Then in your action block, you can do `self?.deletePost(post)`.